doc-examples/src/main/scala/geotrellis/doc/examples/spark/ShardingKeyIndex.scala

```diff
@@ -0,0 +1,128 @@
+package geotrellis.doc.examples.spark
+
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.index._
+import geotrellis.spark.io.json._
+import scala.reflect.ClassTag
+
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+// --- //
+
+/**
+ * An example of a generic KeyIndex which accounts for sharding that could occur
+ * in a data store like Apache Accumulo. Given a shard count (pre-configured),
+ * this index adds a "shard prefix" to the true index as given by the ''inner''
+ * argument. Accumulo can shard based off of the first digit of a row ID,
+ * and since we use a round-robin approach to generate prefixes, this distributes
+ * spatially close indices across different shards, and thus helps avoid hot spots.
+ *
+ * ==Assumptions==
+ *   - The given shard count will be between 1 and 8.
+ *   - The ''inner'' index will produce a value less than 2^60 for any
+ *     given key.
+ */
+class ShardingKeyIndex[K](val inner: KeyIndex[K], val shardCount: Int) extends KeyIndex[K] {
+
+  /* Necessary for extending `KeyIndex` */
+  def keyBounds: KeyBounds[K] =
+    inner.keyBounds
```

Comment:
No, since the Keys are actually completely decoupled from the index side of things, and the shard doesn't effect anything but the index, the keybounds remains valid. So we're good.